core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -189,7 +202,7 @@ public void put(Put put) throws CrudException {
     }
 
     if (put.getCondition().isPresent()) {
-      if (put.isImplicitPreReadEnabled()) {
+      if (put.isImplicitPreReadEnabled() && !snapshot.containsKeyInReadSet(key)) {
```

Comment:
Good point, thank you. It's better to have since we might have a cached result that is acquired by another `Get` different from the `createGet()` one. But yes, we can rewrite it as you suggested.

Reply:
Fixed in 3f72489.